webapp/src/main/resources/public/js/utils/RepositoryInputHelper.js

```diff
@@ -53,4 +61,30 @@ export function flattenRepositoryLocales(repositoryLocales) {
     });
 
     return result;
-}
\ No newline at end of file
+}
+
+// Converts flattened repository locales back to hierarchical structure
+// [{parentLocale, childLocales, toBeFullyTranslated}...] excluding the sourceLocale
+export function unflattenRepositoryLocales(flattened, sourceLocale) {
```

Comment:
Nit: unit tests for data structure transformations would be handy to confirm it works and understanding the mapping that's done (since the comment above has no examples)